src/app/features/courses/courses.component.ts

```diff
@@ -1,20 +1,30 @@
+import { ButtonLabelService } from "@app/services/button-label.service";
 import { Component, OnInit } from "@angular/core";
 import { CoursesStoreService } from "@app/services/courses-store.service";
 import { CourseModel } from "@app/shared/models/course.model";
 import { Observable } from "rxjs";
+import { Router } from "@angular/router";
 
 @Component({
   selector: "app-courses",
   templateUrl: "./courses.component.html",
   styleUrls: ["./courses.component.scss"],
 })
 export class CoursesComponent implements OnInit {
-  constructor(private coursesStoreService: CoursesStoreService) {}
+  constructor(
+    private coursesStoreService: CoursesStoreService,
+    protected buttonLabelService: ButtonLabelService,
+    private router: Router
+  ) {}
   courses$!: Observable<CourseModel[]>;
 
   ngOnInit(): void {
     this.coursesStoreService.getAll();
     this.coursesStoreService.getAllAuthors();
     this.courses$ = this.coursesStoreService.courses$;
   }
+
+  onNewCourseClick(): void {
+    this.router.navigate(["courses/add"]);
```

Comment:
Just poped into my mind, you already have API_ROUTES const exported. Why not use them here? (and other places where you perform a navigation)

Reply:
True, I can do that for most of the cases, but there are some where I can't, such as "courses/edit", "registration" (API endpoint is /register). That's why I thought that if I were to use this pattern, it would make more sense to create a separate one for API routes and navigation routes. Should I implement this separate navigation routes constant?

Reply:
Ohh, I see. Handle what you can by API_ROUTES, and maybe add the exception to some other constant? I just left another comment on this topic.